dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -142,7 +143,7 @@ public T createHosted(@ApiParam(hidden = true) @Auth User user,
      * @param namespace
      * @return
      */
-    protected abstract T getEntry(User user, String registry, String name, String descriptorType, String namespace);
+    protected abstract T getEntry(User user, String registry, String name, String descriptorType, String namespace, String entryName);
```

Comment:
See comment above, this is getting worse. I wonder if we can type some of this, like the registry and/or descriptorType

Document entryname at least

Reply:
Types could be added to registry and desc types.